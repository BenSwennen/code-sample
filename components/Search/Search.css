.root {
  --paddingHorizontal: 30px;
  --paddingVertical: 4px;

  position: relative;

  input {
    font-weight: $fontRegular;
    color: $black;
    padding: var(--paddingVertical) 12px var(--paddingVertical) var(--paddingHorizontal);
    box-sizing: border-box;
    border: none;
    background: transparent;
    border-radius: 5px;
  }

  input::-webkit-input-placeholder {
    color: $gray;
    text-transform: none;
  }

  input:-moz-placeholder {
    color: $gray;
    text-transform: none;
  }

  input:-ms-input-placeholder {
    color: $gray;
    text-transform: none;
  }

  &.borderBottom input {
    border-bottom: 1px solid $borderColor;

    &:focus {
      border-bottom: 1px solid $blue;
    }
  }

  &.border {
    --paddingHorizontal: 36px;
    --paddingVertical: 9px;

    input {
      border: 1px solid $grayLighter;
      border-radius: 4px !important;
    }
  }

  input:focus {
    background: $white;
  }

  &.right > input {
    padding-left: 10px;
    padding-right: 30px;
  }

  svg {
    color: $gray;
    position: absolute;
    top: 50%;
    margin-top: -9px;
    left: calc((var(--paddingHorizontal) / 2) - 6px);
  }

  &.right > svg {
    left: auto;
    right: calc((var(--paddingHorizontal) / 2) - 6px);
  }

  .placeholder {
    font-size: .625rem;
    position: absolute;
    top: 50%;
    right: 12px;
    margin-top: -.375rem;
    user-select: none;

    label {
      cursor: text;
      color: $grayLight;
    }
  }

  &.right > .placeholder {
    display: none;
  }
}
