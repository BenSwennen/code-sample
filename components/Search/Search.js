import React from "react";
import PropTypes from "prop-types";
import cx from "classnames";
import SearchIcon from "rialto-icons/Search";
import styles from "./Search.css";

const Search = (props) => {
  const {
    isBordered,
    isBottomBordered,
    isRight,
    onSearch,
    placeholder,
    placeholderRight,
    value,
  } = props;

  return (
    <div
      className={cx(styles.root, {
        [styles.border]: isBordered,
        [styles.borderBottom]: isBottomBordered,
        [styles.right]: isRight,
      })}
    >
      <SearchIcon />
      <input
        id="search"
        type="text"
        value={value}
        placeholder={placeholder}
        onChange={evt => onSearch(evt.target.value)}
      />
      {placeholderRight && !value &&
        <div className={styles.placeholder}>
          <label htmlFor="search">{placeholderRight}</label>
        </div>
      }
    </div>
  );
};

Search.propTypes = {
  isBordered: PropTypes.bool,
  isBottomBordered: PropTypes.bool,
  onSearch: PropTypes.func.isRequired,
  placeholder: PropTypes.string.isRequired,
  placeholderRight: PropTypes.string,
  isRight: PropTypes.bool,
  value: PropTypes.string,
};

Search.defaultProps = {
  value: "",
};

export default Search;
