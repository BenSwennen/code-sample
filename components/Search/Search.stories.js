import React from "react";
import { storiesOf } from "@storybook/react";
import { action } from "@storybook/addon-actions";
import Search from ".";

storiesOf("Search", module)

  .add("default", () => (
    <Search placeholder="Search" onSearch={action("Search")} />
  ))

  .add("Border Bottom", () => (
    <Search placeholder="Search" onSearch={action("Search")} isBottomBordered />
  ))

  .add("Full Border", () => (
    <Search placeholder="Search" onSearch={action("Search")} isBordered />
  ))

  .add("Icon Right", () => (
    <Search placeholder="Search" onSearch={action("Search")} isRight />
  ))

  .add("Placeholder Right", () => (
    <Search
      placeholder="Search"
      placeholderRight="(name, kind, tag..)"
      onSearch={action("Search")}
      isBordered
    />
  ));
