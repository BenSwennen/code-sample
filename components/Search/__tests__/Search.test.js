import React from "react";
import renderer from "react-test-renderer";
import Search from "../";

describe("<Search />", () => {
  it("renders correctly", () => {
    const mockFn = jest.fn();
    const component = renderer.create(
      <Search placeholder="Search" onSearch={mockFn} isBorder isRight />
    ).toJSON();
    expect(component).toMatchSnapshot();
  });
});
