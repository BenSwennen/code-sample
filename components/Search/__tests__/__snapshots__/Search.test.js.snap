// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> renders correctly 1`] = `
<div
  className="root right"
>
  <svg
    style={
      Object {
        "display": "inline-block",
        "fill": "true",
        "height": "1rem",
        "width": "1rem",
      }
    }
  >
    <svg
      viewBox="0 0 17 16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        fill="none"
        fillRule="evenodd"
      >
        <path
          d="M11.165 9.043A5.962 5.962 0 0 0 12 6a6 6 0 1 0-6 6c1.112 0 2.15-.308 3.043-.835l4.396 4.396a1.5 1.5 0 1 0 2.122-2.121l-4.396-4.397z"
          fill="#91A6B6"
        />
        <path
          d="M1 6a5 5 0 1 1 10 0A5 5 0 0 1 1 6z"
          fill="#FFF"
          fillOpacity=".7"
        />
        <path
          d="M9 6a3 3 0 0 0-3-3V2a4 4 0 0 1 4 4H9z"
          fill="#FFF"
        />
      </g>
    </svg>
  </svg>
  <input
    id="search"
    onChange={[Function]}
    placeholder="Search"
    type="text"
    value=""
  />
</div>
`;
